lib/glific/communications/message.ex

```diff
@@ -30,11 +30,18 @@ defmodule Glific.Communications.Message do
   @doc """
   Send message to receiver using define provider.
   """
-  @spec send_message(Message.t()) :: {:ok, Message.t()}
+  @spec send_message(Message.t()) :: {:ok, Message.t()} | {:error, String.t()}
   def send_message(message) do
     message = Repo.preload(message, [:receiver, :sender, :media])
-    apply(provider_module(), @type_to_token[message.type], [message])
-    {:ok, Communications.publish_data(message, :sent_message)}
+
+    case Contacts.can_send_message_to?(message.receiver) do
+      true ->
+        apply(provider_module(), @type_to_token[message.type], [message])
+        {:ok, Communications.publish_data(message, :sent_message)}
```

Comment:
Shouldt we replace this with if/else since its a true/false scenario

Reply:
Yes, I was also confused between these two approaches. 
I will make that change now.